### PR TITLE
Corrects typespec, corrects credo issues, and updates dependencies

### PR DESCRIPTION
**Pull request by britton-jb**

### CODE REVIEW

File: lib/task_bunny/job.ex [hunk 1]
```diff
@@ -85,7 +85,7 @@ defmodule TaskBunny.Job do
       end
 
   """
-  @callback perform(any) :: :ok | {:ok, any} | {:error, term}
+  @callback perform(any) :: :ok | {:ok, any} | {:error, term} | {:reject, any} | :reject
```
Comment (britton-jb) [inline lib/task_bunny/job.ex:88]: This was the spec @keatz55 and I ran into issues with.

**Comment by britton-jb** [comment]
The low hanging fruit left is opening up Credo to run on the test files, and addressing the compiler warnings around stacktrace being deprecated.

**Review by hammeraj — APPROVED** [review]

**Comment by hammeraj** [comment]
yeah a github action to automate the testing and such shouldn't be too bad, can probably pull in a rmq container for running tests against too

**Comment by britton-jb** [comment]
That would be good for the future.

**Comment by hammeraj** [comment]
Also a friendly reminder to tag 0.4.2

**Pull request merged**